core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -1630,6 +1664,12 @@
             #'msg-password nil
 
             (cmd-send-error conn (err-protocol-violation "unknown client message"))))))
+    (catch ExceptionInfo e
```

Comment:
Not sure this is actually the right place to catch things. Probably good to talk through a general error handling approach for these kind of errors.